query/command_graphite_test.go

```diff
@@ -0,0 +1,350 @@
+			expected: []api.Timeseries{
+				{
+					Values: []float64{1, 2, 3},
+					TagSet: api.TagSet{
+						"app":      "server",
+						"dc":       "north",
+						"quantity": "mean",
+					},
+				},
+				{
+					Values: []float64{4, 5, 6},
+					TagSet: api.TagSet{
+						"app":      "server",
+						"dc":       "north",
+						"quantity": "median",
+					},
+				},
+				{
+					Values: []float64{7, 8, 9},
+					TagSet: api.TagSet{
+						"app":      "server",
+						"dc":       "south",
+						"quantity": "mean",
+					},
+				},
+				{
+					Values: []float64{2, 4, 8},
+					TagSet: api.TagSet{
+						"app":      "server",
+						"dc":       "south",
+						"quantity": "median",
+					},
+				},
+				{
+					Values: []float64{1, 1, 1},
+					TagSet: api.TagSet{
+						"app":      "proxy",
+						"dc":       "south",
+						"quantity": "mean",
+					},
+				},
+				{
+					Values: []float64{2, 2, 2},
+					TagSet: api.TagSet{
+						"app":      "proxy",
+						"dc":       "south",
+						"quantity": "median",
+					},
+				},
+			},
+		},
+		{
+			query: "graphite('latency.%host%.%method%') from -10m to now",
+			expected: []api.Timeseries{
+				{
+					Values: []float64{3, 3, 3},
+					TagSet: api.TagSet{
+						"host":   "host45",
+						"method": "http",
+					},
+				},
+				{
+					Values: []float64{4, 4, 4},
+					TagSet: api.TagSet{
+						"host":   "host68",
+						"method": "http",
+					},
+				},
+				{
+					Values: []float64{5, 5, 5},
+					TagSet: api.TagSet{
+						"host":   "host68",
+						"method": "rpc",
+					},
+				},
+				{
+					Values: []float64{6, 6, 6},
+					TagSet: api.TagSet{
+						"host":   "host22",
+						"method": "rpc",
+					},
+				},
+			},
+		},
+		{
+			query: "graphite('latency.%host%.%method%') where method = 'rpc' from -10m to now",
+			expected: []api.Timeseries{
+				{
+					Values: []float64{5, 5, 5},
+					TagSet: api.TagSet{
+						"host":   "host68",
+						"method": "rpc",
+					},
+				},
+				{
+					Values: []float64{6, 6, 6},
+					TagSet: api.TagSet{
+						"host":   "host22",
+						"method": "rpc",
+					},
+				},
+			},
+		},
+	}
+	for _, test := range tests {
+		command, err := Parse(test.query)
+		if err != nil {
+			t.Fatalf("Unexpected error while parsing")
+			return
+		}
+		if command.Name() != "select" {
+			t.Errorf("Expected select command but got %s", command.Name())
+			continue
+		}
+		rawResult, err := command.Execute(ExecutionContext{Backend: fakeBackend, API: fakeAPI, FetchLimit: 1000, Timeout: 0})
+		if err != nil {
+			t.Errorf("Unexpected error while execution: %s", err.Error())
+			continue
+		}
+		seriesListList, ok := rawResult.([]api.SeriesList)
+		if !ok || len(seriesListList) != 1 {
+			t.Errorf("expected query `%s` to produce []value; got %+v :: %T", test.query, rawResult, rawResult)
+			continue
+		}
+		list := seriesListList[0]
+		if err != nil {
+			t.Fatal(err)
+		}
+		a := assert.New(t)
+		expectedSeries := test.expected
```

Comment:
Should check the number of series you get back is the same as the number of series you expect.